tests/bytecode.rs

```diff
@@ -0,0 +1,73 @@
+use std::str::FromStr;
```

Comment:
I think this is cleaner as a 
```
#[cfg(test)]
mod tests {
    use super::*;
}
```

in `storage.rs`. It's more appropriate for unit tests for location, removing the need to make fields `pub`, etc. Most stuff in `tests` are integration tests for the `execute_*` functions.

Reply:
Good call, not sure what I was thinking here. These are definitely unit tests I'm targetting.